### PR TITLE
Fix/15373 last message in the report shows for a brief moment

### PR DESCRIPTION
**Pull request by tienifr**
### Fixed Issues

$ https://github.com/Expensify/App/issues/15373
$ https://github.com/Expensify/App/issues/15373#issuecomment-1452061630

### Tests

Sample password protected PDF: [password-protected.pdf](https://github.com/Expensify/App/files/11043319/password-protected.pdf)
Password: 1234

1. Sign in to NewExpensify Chrome on an Android
2. Select a chat
3. Send a password protected pdf
4. Click on password protected pdf
5. Click on blue hyperlink to "enter password"
6. As the keyboard slides up, verify that you will not see the chat behind.
- [x] Verify that no errors appear in the JS console

### QA Steps

Sample password protected PDF: [password-protected.pdf](https://github.com/Expensify/App/files/11043319/password-protected.pdf)
Password: 1234

1. Sign in to NewExpensify Chrome on an Android
2. Select a chat
3. Send a password protected pdf
4. Click on password protected pdf
5. Click on blue hyperlink to "enter password"
6. As the keyboard slides up, verify that you will not see the chat behind.
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/113963320/226530433-076bb9be-4a10-436e-8582-c5ab4840f225.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/113963320/226530404-3d130267-2977-4a4c-8f17-f9ab0b68cc43.mov

 
</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/113963320/226530388-7e96219c-f54a-432b-94bf-d8c8f9b2d5e1.mov

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/113963320/226530374-72d2c505-1c16-46c4-a1a4-d3b5212f30af.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/113963320/226530360-023278bd-194e-4a4b-bbf0-a6b693f97ef2.mov

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/113963320/226530346-03727faf-299a-4b5c-912f-607e8408c42f.mov

</details>

### CODE REVIEW

File: src/libs/getSafeAreaPaddingTop/index.android.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import {StatusBar} from 'react-native';
+
+export default function getSafeAreaPaddingTop() {
+    return StatusBar.currentHeight || 0;
```
Comment (situchan) [inline src/libs/getSafeAreaPaddingTop/index.android.js:4]: `StatusBar.currentHeight` value always exists in android so I think `|| 0` is redundant.

**Review by situchan — COMMENTED** [review]
@tienifr did you push all your changes? It's not complete yet.
image

File: src/styles/StyleUtils.js [hunk 2]
```diff
@@ -129,7 +130,7 @@ function getDefaultWorspaceAvatarColor(workspaceName) {
  */
 function getSafeAreaPadding(insets) {
```
Comment (situchan) [inline src/styles/StyleUtils.js:131]:
```suggestion
function getSafeAreaPadding(insets, statusBarTranslucent) {
```

File: src/styles/StyleUtils.js [hunk 3]
```diff
@@ -129,7 +130,7 @@ function getDefaultWorspaceAvatarColor(workspaceName) {
...
     return {
-        paddingTop: insets.top,
+        paddingTop: getSafeAreaPaddingTop(insets),
```
Comment (situchan) [inline src/styles/StyleUtils.js:133]:
```suggestion
        paddingTop: getSafeAreaPaddingTop(insets, statusBarTranslucent),
```

File: src/libs/getSafeAreaPaddingTop/index.android.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import {StatusBar} from 'react-native';
+
+export default function getSafeAreaPaddingTop() {
```
Comment (situchan) [inline src/libs/getSafeAreaPaddingTop/index.android.js:3]:
```suggestion
export default function getSafeAreaPaddingTop(insets, statusBarTranslucent) {
```

File: src/libs/getSafeAreaPaddingTop/index.android.js (hunk 1 above)
Comment (situchan) [inline src/libs/getSafeAreaPaddingTop/index.android.js:4]:
```suggestion
    return (statusBarTranslucent && StatusBar.currentHeight) || 0;
```

**Review by situchan — CHANGES_REQUESTED** [review]

**Comment by situchan** [comment]
@tienifr I think I gave all my feedback. Please check suggested changes.

**Comment by tienifr** [comment]
@situchan Thank you for suggesting. I have updated the code.
@aimane-chnaif @aldo-expensify After the PR is fully tested by @situchan, please help me review it!

**Comment by situchan** [comment]
> I verified the JSDocs style guidelines (in [STYLE.md](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed

Let's follow jsdocs for new functions

**Review by aimane-chnaif — CHANGES_REQUESTED** [review]

**Comment by tienifr** [comment]
@situchan @aimane-chnaif updated, thanks for the feedback

**Comment by situchan** [comment]
https://user-images.githubusercontent.com/108292595/226704604-eee09157-0b5f-4e4d-89fd-930a635c9631.mp4

While testing, I noticed weird bugs on password protected pdf preview:
- Loading forever sometimes (I think this is already reported). After close and open again, works
- After click `enter the password`, password input modal briefly appears and disappears

I confirmed that this happens on `main`, not introduced by this PR.
Maybe regression from carousel PR.

**Comment by situchan** [comment]
Except the issue above, works well on all devices / platforms.

### Screenshots/Videos
<details>
<summary>Web</summary>


https://user-images.githubusercontent.com/108292595/226713118-180151c7-bfa1-4877-bfb6-bd040ab8fe17.mov



</details>

<details>
<summary>Mobile Web - Chrome</summary>


https://user-images.githubusercontent.com/108292595/226713182-99edec9f-03ab-4db9-a026-d9148ee3e31f.mp4



</details>

<details>
<summary>Mobile Web - Safari</summary>


https://user-images.githubusercontent.com/108292595/226713241-ec2c56f1-d9e9-4d2a-b519-52d039ab8bd0.mp4



</details>

<details>
<summary>Desktop</summary>


https://user-images.githubusercontent.com/108292595/226713278-6b85b491-59b0-47f2-92a4-2cc3be055414.mov



</details>

<details>
<summary>iOS</summary>


https://user-images.githubusercontent.com/108292595/226713309-850e4526-532c-488e-97f9-f6243a99e72d.mp4



</details>

<details>
<summary>Android</summary>


https://user-images.githubusercontent.com/108292595/226713448-666d8391-3386-4e43-b4c6-ebd8f767bee3.mp4


https://user-images.githubusercontent.com/108292595/226713569-456afd5d-2f66-4664-8133-e7b66f0e9164.mp4



</details>

**Comment by situchan** [comment]
> After click `enter the password`, password input modal briefly appears and disappears

I reported this bug [here](https://expensify.slack.com/archives/C049HHMV9SM/p1679427115759159)

**Review by aimane-chnaif — APPROVED** [review]
Looks good and tests well on all kinds of modals.

**Comment by tienifr** [comment]
Hi @aldo-expensify We have the full test from @situchan and the approval of @aimane-chnaif. Please help me review the PR. Thank you!

**Review by situchan — APPROVED** [review]

**Comment by aldo-expensify** [comment]
Thanks, I'll review today

**Review by aldo-expensify — APPROVED** [review]
LGTM, tested well.

**Comment by aldo-expensify** [comment]
Thank you all for the collaborative work :)

**Pull request merged**